src/SqlAsyncEnumerable.cs

```diff
@@ -77,8 +77,14 @@ public SqlAsyncEnumerator(SqlConnection connection, SqlAttribute attribute)
             public ValueTask DisposeAsync()
             {
                 // Doesn't seem like there's an async version of closing the reader/connection
-                this._reader.Close();
-                this._connection.Close();
+                if (this._reader != null)
+                {
+                    this._reader.Close();
+                }
+                if (this._connection != null && this._connection.State == System.Data.ConnectionState.Open)
```

Comment:
Why are you adding these checks? We create the connection in the constructor so that should never be null, and closing an already-closed connection just no-ops so we should be fine to just leave it like it is currently.

Reply:
Forgot to remove this (was debugging and it's throwing null reference, so added all these)